tests/nlu/data_augmentation/test_augmenter.py

```diff
@@ -37,167 +35,202 @@ def test_augmenter_create_tokenizer(run: Callable[..., RunResult],):
-        intents_to_augment=intents_to_augment,
-        min_paraphrase_sim_score=0.0,
-        max_paraphrase_sim_score=0.99,
+    paraphrases = rasa.shared.nlu.training_data.loading.load_data(
+        "data/test_nlu_paraphrasing/paraphrases.yml"
     )
-    assert len(pool["check_earnings"]) == 18
-    assert len(pool["check_recipients"]) == 18
-    assert len(pool["ask_transfer_charge"]) == 6
 
     pool = _create_paraphrase_pool(
         paraphrases=paraphrases,
         intents_to_augment=intents_to_augment,
         min_paraphrase_sim_score=0.0,
-        max_paraphrase_sim_score=0.45,
+        max_paraphrase_sim_score=1.0,
     )
 
-    assert len(pool["check_earnings"]) == 3
-    assert len(pool["check_recipients"]) == 8
-    assert len(pool["ask_transfer_charge"]) == 2
-
-    pool = _create_paraphrase_pool(
-        paraphrases=paraphrases,
-        intents_to_augment=intents_to_augment,
-        min_paraphrase_sim_score=0.4,
-        max_paraphrase_sim_score=0.5,
-    )
-    assert len(pool["check_earnings"]) == 5
-    assert len(pool["check_recipients"]) == 8
-    assert len(pool["ask_transfer_charge"]) == 1
-
-
-def test_augmenter_augmentation_factor_resolution(run: Callable[..., RunResult],):
+    assert len(pool) == len(intents_to_augment)
+    assert all([intent in pool for intent in intents_to_augment])
+
+
+@pytest.mark.parametrize(
+    "augmentation_factor, intent, size_after_augmentation",
+    [
+        (0.0, "check_recipients", None),
+        (0.0, "inform", None),
+        (0.0, "search_transactions", None),
+        (0.0, "check_balance", None),
+        (0.0, "transfer_money", None),
+        (0.1, "check_recipients", None),
+        (0.1, "inform", None),
+        (0.1, "search_transactions", None),
+        (0.1, "check_balance", None),
+        (0.1, "transfer_money", None),
+        (0.5, "check_recipients", None),
+        (0.5, "inform", 2),
+        (0.5, "search_transactions", None),
+        (0.5, "check_balance", None),
+        (0.5, "transfer_money", None),
+        (1.0, "check_recipients", 1),
+        (1.0, "inform", 4),
+        (1.0, "search_transactions", 1),
+        (1.0, "check_balance", 1),
+        (1.0, "transfer_money", 1),
+        (2.0, "check_recipients", 2),
+        (2.0, "inform", 8),
+        (2.0, "search_transactions", 2),
+        (2.0, "check_balance", 2),
+        (2.0, "transfer_money", 2),
+        (5.0, "check_recipients", 5),
+        (5.0, "inform", 20),
+        (5.0, "search_transactions", 5),
+        (5.0, "check_balance", 5),
+        (5.0, "transfer_money", 5),
+        (-1.0, "check_recipients", None),
+        (-1.0, "inform", None),
+        (-1.0, "search_transactions", None),
+        (-1.0, "check_balance", None),
+        (-1.0, "transfer_money", None),
+        (-666, "check_recipients", None),
+        (-666, "inform", None),
+        (-666, "search_transactions", None),
+        (-666, "check_balance", None),
+        (-666, "transfer_money", None),
```

Comment:
Should we even allow negative augmentation factor? If it helps to reduce the complexity we should just raise an error if augmentation factor is negative in the command line parameters itself. Wdyt?

Reply:
At the moment we have the option to pass `-1` (or any other negative number) to mean "use all the examples". 

Do you think its better to remove this option?

Reply:
Yeah, I think it's better to start without it. We can add it in if users request for it explicitly.